src/clients/c++/request_common.h

```diff
@@ -155,7 +155,7 @@ class InputImpl : public InferContext::Input {
   bool IsSharedMemory() const { return (io_type_ == SHARED_MEMORY); }
   const std::string& GetSharedMemoryName() const { return shm_name_; }
   size_t GetSharedMemoryOffset() const { return shm_offset_; }
-  size_t GetSharedMemoryByteSize() const { return byte_size_; }
+  size_t GetSharedMemoryByteSize() const { return shm_byte_size_; }
```

Comment:
This looks like a bug in the input shared memory...
But we shouldn't really have shm_byte_size_ at all... instead in SetSharedMemory we just need to validate that the shared memory size == byte_size_ like is done in SetRaw.
You should put this fix in a different PR so we can get it submitted ASAP since it is an important change.

Reply:
Will do this ASAP

Reply:
Resolved by #541